http.go

```diff
@@ -275,13 +275,11 @@ func (hc *HttpCall) ResponseBodyContains(value string) Step {
 // ResponseBodyMatches is a Step that when executed ensures there is
 // a non-nil hc.ResponseBody and errors unless the hc.ResponseBody
 // matches the regular expression parameter.
-func (hc *HttpCall) ResponseBodyMatches(pattern string) Step {
-	return NewNamedStep("ResponseBodyMatches", func() error {
+func (hc *HttpCall) ResponseBodyMatches(pattern *regexp.Regexp) Step {
+	return NewNamedStep(fmt.Sprintf("ResponseBodyMatches(%s)", pattern), func() error {
```

Comment:
Apologies - I've just done some digging and this does not error which surprises me - just been reading the docs on `fmt` and it's more subtle than I thought.

```package main
import "fmt"

type Thing struct{}
func (t *Thing) String() string {
	return "weird"
}

func main() {
	fmt.Println(fmt.Sprintf("%s", &Thing{}))
}
```

I assumed the above would error as Thing is not a string. But it turns out there are special rules that allow this to work. I'm afraid I would still prefer `%v` though, but I apologise for suggesting you'd not tested it.